### DISCUSSION TITLE
Append address extension to a domain loaded in from fixture

### DISCUSSION THREAD

**Discussion by Alex-passiv** [cypress-io/cypress#14561] [Component Testing]
Hi, I have created a file where I am loading in a fixture domain, this is in order to have cleaner tests and be able to test locally, beta and on prod without changing anything except for one fixture. 

My question is, for pages that are extensions of the main domain, any ideas how i would append the extension to the read in fixture data? Is this even possible? 

Here is my code:
` cy.fixture('testDomain').as('server')
    cy.get('@server').then(domain => {
    cy.visit(domain.test) + "/app/help" })`

currently it doesnt work like this but I kept it this way to make it easier to visualize what I want. 

Cheers.

**Comment by tit** [comment]
You can use [cypress.env.json](https://docs.cypress.io/guides/guides/environment-variables.html#Option-2-cypress-env-json) files.

Create some files for each environments, like:

- `cypress.env.qa.json`
- `cypress.env.localhost.json`
- `cypress.env.production.json`

with content:

```
// cypress.env.qa.json

{
  BASE_URL: "https://qa.example.com"
}
```

```
// cypress.env.localhost.json

{
  BASE_URL: "https://localhost:8080"
}
```

Add next code in `/cypress/plugin/index.js` file:

```
// /cypress/plugin/index.js
module.exports = (on, config) => {
  config.baseUrl = config.env.INCOUNTRY_URL;
}
```

Before run tests, execute: `cp ./cypress.env.localhost.json ./cypress.env.json` in test root directory.
For example, you can add this script in `package.json`:

```
{
  "scripts": {
    "env:set:production": "cp ./cypress.env.localhost.json ./cypress.env.json",
  }
}
```

In tests you can use relative path, like `cy.visit('/foo')` for `https://qa.example.com/foo` url.
